bench/bm_buffer.cpp

```diff
@@ -70,7 +74,19 @@ testNewAPI(Buffer auto &buffer, const std::size_t vector_length, const std::size
                 std::size_t nSamplesProduced = 0;
                 barrier.arrive_and_wait();
                 while (nSamplesProduced <= (min_samples + nProducer - 1) / nProducer) {
-                    writer.publish([](auto &) {}, vector_length);
+                    if constexpr (PublisherAPI == WriteApi::via_lambda) {
+                        writer.publish([](auto &) {}, vector_length);
+                    } else if constexpr (PublisherAPI == WriteApi::via_split_get_publish) {
+                        auto [data, token] = writer.get(vector_length);
+
+                        writer.publish(token, vector_length);
+                    }  else if constexpr (PublisherAPI == WriteApi::via_split_get_publish_RAII) {
+                        auto [data, token] = writer.get_alt(vector_length);
+
+                        token.publish(vector_length);
```

Comment:
I'd prefer a more descriptive name than `get` here. You're getting something very *specific*; not like `std::get(tuple)` which could be *anything*. So the name should be specific. How would you write it in prose? "Request to publish `vector_length` samples via `writer`. Write samples. Finalize/commit the publication."?
I also don't feel too happy about the `std::pair` return type. However, if the function name suggests that it returns two objects, maybe that'd make me happier. `writer.tell_me_where_can_I_put_my_samples_and_how_can_I_let_you_know_when_I_am_done` :stuck_out_tongue_winking_eye: 
```c++
auto bucket = writer.request_publication_bucket(vector_length);
for (auto& x : bucket) {
  x = 1;
}
bucket.publish(vector_length);
```
Can't we instead ask for one thing that *is a* `span` or contains a `span` or simply `is a` `contiguous_range`. And at the same it is the RAII object that ensures the request isn't forgotten?